src/encoder.rs

```diff
@@ -42,7 +42,13 @@ pub struct Frame {
 const FRAME_MARGIN: usize = 16 + SUBPEL_FILTER_SIZE;
 
 impl Frame {
-    pub fn new(width: usize, height:usize) -> Frame {
+    pub fn new(width: usize, height: usize, chroma_sampling: ChromaSampling) -> Frame {
+        let (chroma_width, chroma_height) = match chroma_sampling {
```

Comment:
This match should probably be extracted into a method on `ChromaSampling`, since you're using the same logic in two places.